src/dir.targets

```diff
@@ -178,6 +203,28 @@
     <Error Condition="'$(TestRunExitCode)' != '0'" Text="One or more tests failed while running tests from '$(MSBuildProjectName)' please check log for details!" />
   </Target>
 
+  <Target Name="CheckTestCategories"
+          AfterTargets="Test"
+          Condition="'$(TestDisabled)'!='True'">
+
+    <ItemGroup>
+      <!-- Intersection of TestCategoriesItems and RunTestsWithCategoriesItems -->
+      <RunTestsCategoriesFiltered Include="@(RunTestsWithCategoriesItems)" Condition="'@(RunTestsWithCategoriesItems)'=='@(TestCategoriesItems)' And '%(Identity)'!=''" />
+      <RunTestsCategoriesFiltered Remove="@(RunTestsWithoutCategoriesItems)" />
```

Comment:
As we chatted can you switch this to using "%(...)" in the condition so it is a little clearer what it is trying to do.